### DISCUSSION TITLE
NextJS Build Error

### DISCUSSION THREAD

**Discussion by austin495** [vercel/next.js#69237] [Help] [answered]
The below is my code when I run `npx next build` is through the error. Please help me to fix this issue I have tried many things but didn't help. Thanks!

Code:
`'use client'
import { Badge } from "@/components/ui/badge"
import { Button } from "@/components/ui/button"
import { Calendar } from "@/components/ui/calendar"
import { Card, CardContent, CardDescription, CardHeader, CardTitle, } from "@/components/ui/card"
import { ChartConfig, ChartContainer, ChartTooltip, ChartTooltipContent, } from "@/components/ui/chart"
import { Popover, PopoverContent, PopoverTrigger, } from "@/components/ui/popover"
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue, } from "@/components/ui/select"
import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow, } from "@/components/ui/table"
import { cn } from "@/lib/utils"
import { addDays, format } from "date-fns"
import { Activity, Calendar as CalendarIcon, CreditCard, DollarSign, Loader2, Users } from "lucide-react"
import * as React from "react"
import { DateRange } from "react-day-picker"
import { Bar, BarChart, CartesianGrid, XAxis } from "recharts"

const chartData = [
  { date: "2024-04-01", google: 222, facebook: 150 },
  { date: "2024-04-02", google: 97, facebook: 180 },
  { date: "2024-04-03", google: 167, facebook: 120 },
  { date: "2024-04-04", google: 242, facebook: 260 },
  { date: "2024-04-05", google: 373, facebook: 290 },
  { date: "2024-04-06", google: 301, facebook: 340 },
  { date: "2024-04-07", google: 245, facebook: 180 },
  { date: "2024-04-08", google: 409, facebook: 320 },
  { date: "2024-04-09", google: 59, facebook: 110 },
  { date: "2024-04-10", google: 261, facebook: 190 },
  { date: "2024-04-11", google: 327, facebook: 350 },
  { date: "2024-04-12", google: 292, facebook: 210 },
  { date: "2024-04-13", google: 342, facebook: 380 },
  { date: "2024-04-14", google: 137, facebook: 220 },
  { date: "2024-04-15", google: 120, facebook: 170 },
  { date: "2024-04-16", google: 138, facebook: 190 },
  { date: "2024-04-17", google: 446, facebook: 360 },
  { date: "2024-04-18", google: 364, facebook: 410 },
  { date: "2024-04-19", google: 243, facebook: 180 },
  { date: "2024-04-20", google: 89, facebook: 150 },
  { date: "2024-04-21", google: 137, facebook: 200 },
  { date: "2024-04-22", google: 224, facebook: 170 },
  { date: "2024-04-23", google: 138, facebook: 230 },
  { date: "2024-04-24", google: 387, facebook: 290 },
  { date: "2024-04-25", google: 215, facebook: 250 },
  { date: "2024-04-26", google: 75, facebook: 130 },
  { date: "2024-04-27", google: 383, facebook: 420 },
  { date: "2024-04-28", google: 122, facebook: 180 },
  { date: "2024-04-29", google: 315, facebook: 240 },
  { date: "2024-04-30", google: 454, facebook: 380 },
  { date: "2024-05-01", google: 165, facebook: 220 },
  { date: "2024-05-02", google: 293, facebook: 310 },
  { date: "2024-05-03", google: 247, facebook: 190 },
  { date: "2024-05-04", google: 385, facebook: 420 },
  { date: "2024-05-05", google: 481, facebook: 390 },
  { date: "2024-05-06", google: 498, facebook: 520 },
  { date: "2024-05-07", google: 388, facebook: 300 },
  { date: "2024-05-08", google: 149, facebook: 210 },
  { date: "2024-05-09", google: 227, facebook: 180 },
  { date: "2024-05-10", google: 293, facebook: 330 },
  { date: "2024-05-11", google: 335, facebook: 270 },
  { date: "2024-05-12", google: 197, facebook: 240 },
  { date: "2024-05-13", google: 197, facebook: 160 },
  { date: "2024-05-14", google: 448, facebook: 490 },
  { date: "2024-05-15", google: 473, facebook: 380 },
  { date: "2024-05-16", google: 338, facebook: 400 },
  { date: "2024-05-17", google: 499, facebook: 420 },
  { date: "2024-05-18", google: 315, facebook: 350 },
  { date: "2024-05-19", google: 235, facebook: 180 },
  { date: "2024-05-20", google: 177, facebook: 230 },
  { date: "2024-05-21", google: 82, facebook: 140 },
  { date: "2024-05-22", google: 81, facebook: 120 },
  { date: "2024-05-23", google: 252, facebook: 290 },
  { date: "2024-05-24", google: 294, facebook: 220 },
  { date: "2024-05-25", google: 201, facebook: 250 },
  { date: "2024-05-26", google: 213, facebook: 170 },
  { date: "2024-05-27", google: 420, facebook: 460 },
  { date: "2024-05-28", google: 233, facebook: 190 },
  { date: "2024-05-29", google: 78, facebook: 130 },
  { date: "2024-05-30", google: 340, facebook: 280 },
  { date: "2024-05-31", google: 178, facebook: 230 },
  { date: "2024-06-01", google: 178, facebook: 200 },
  { date: "2024-06-02", google: 470, facebook: 410 },
  { date: "2024-06-03", google: 103, facebook: 160 },
  { date: "2024-06-04", google: 439, facebook: 380 },
  { date: "2024-06-05", google: 88, facebook: 140 },
  { date: "2024-06-06", google: 294, facebook: 250 },
  { date: "2024-06-07", google: 323, facebook: 370 },
  { date: "2024-06-08", google: 385, facebook: 320 },
  { date: "2024-06-09", google: 438, facebook: 480 },
  { date: "2024-06-10", google: 155, facebook: 200 },
  { date: "2024-06-11", google: 92, facebook: 150 },
  { date: "2024-06-12", google: 492, facebook: 420 },
  { date: "2024-06-13", google: 81, facebook: 130 },
  { date: "2024-06-14", google: 426, facebook: 380 },
  { date: "2024-06-15", google: 307, facebook: 350 },
  { date: "2024-06-16", google: 371, facebook: 310 },
  { date: "2024-06-17", google: 475, facebook: 520 },
  { date: "2024-06-18", google: 107, facebook: 170 },
  { date: "2024-06-19", google: 341, facebook: 290 },
  { date: "2024-06-20", google: 408, facebook: 450 },
  { date: "2024-06-21", google: 169, facebook: 210 },
  { date: "2024-06-22", google: 317, facebook: 270 },
  { date: "2024-06-23", google: 480, facebook: 530 },
  { date: "2024-06-24", google: 132, facebook: 180 },
  { date: "2024-06-25", google: 141, facebook: 190 },
  { date: "2024-06-26", google: 434, facebook: 380 },
  { date: "2024-06-27", google: 448, facebook: 490 },
  { date: "2024-06-28", google: 149, facebook: 200 },
  { date: "2024-06-29", google: 103, facebook: 160 },
  { date: "2024-06-30", google: 446, facebook: 400 },
]

const chartConfig: ChartConfig = {
  views: {
    label: "Leads",
    icon: undefined,
    color: undefined,
    theme: undefined as never,
  },
  google: {
    label: "Google",
    color: "hsl(var(--chart-1))",
  },
  facebook: {
    label: "Facebook",
    color: "hsl(var(--chart-2))",
  },
  themeExample: {
    label: "Theme Based",
    theme: {
      light: "#fff",
      dark: "#000",
    },
    color: undefined as never, // Explicitly setting to never since theme is present
  }
};

export default function Home({
  className,
}: React.HTMLAttributes<HTMLDivElement>) {
  const [date, setDate] = React.useState<DateRange | undefined>({
    from: new Date(2022, 0, 20),
    to: addDays(new Date(2022, 0, 20), 20),
  })
  // const [loading, setLoading] = React.useState(false);
  const [activeChart, setActiveChart] =
    React.useState<keyof typeof chartConfig>("google")

  const total = React.useMemo(
    () => ({
      google: chartData.reduce((acc, curr) => acc + curr.google, 0),
      facebook: chartData.reduce((acc, curr) => acc + curr.facebook, 0),
    }),
    []
  )

  return (
    <div className="flex min-h-screen w-full flex-col">
      <div className="flex p-4 md:gap-8 md:p-8">
        <div className={cn("grid gap-2", className)}>
          <Popover>
            <PopoverTrigger asChild>
              <Button
                id="date"
                variant={"outline"}
                className={cn(
                  "text-dark300_light900 w-[300px] justify-start text-left font-normal",
                  !date && "text-muted-foreground"
                )}
              >
                <CalendarIcon className="mr-2 size-4" />
                {date?.from ? (
                  date.to ? (
                    <>
                      {format(date.from, "LLL dd, y")} -{" "}
                      {format(date.to, "LLL dd, y")}
                    </>
                  ) : (
                    format(date.from, "LLL dd, y")
                  )
                ) : (
                  <span>Pick a date</span>
                )}
              </Button>
            </PopoverTrigger>
            <PopoverContent className="w-auto bg-white p-0" align="start">
              <Calendar
                initialFocus
                mode="range"
                defaultMonth={date?.from}
                selected={date}
                onSelect={setDate}
                numberOfMonths={2}
              />
            </PopoverContent>
          </Popover>
        </div>

        <div>
          <Select>
            <SelectTrigger className="text-dark300_light900 w-[180px]">
              <SelectValue placeholder="Google Ad Account" />
            </SelectTrigger>
            <SelectContent className="bg-white">
              <SelectItem value="google-ad-account">Google Ad Account</SelectItem>
              <SelectItem value="facebook-ad-account">Facebook Ad Account</SelectItem>
            </SelectContent>
          </Select>
        </div>

        <div>
          <Button
            disabled
            variant="outline"
            className="text-dark300_light900"
          >
            <Loader2 className="text-dark300_light900 mr-2 size-4 animate-spin" />
            Please wait
          </Button>
        </div>
      </div>
      <main className="flex flex-1 flex-col gap-4 p-4 md:gap-8 md:p-8">
        <div className="grid gap-4 md:grid-cols-2 md:gap-8 lg:grid-cols-4">
          <Card x-chunk="dashboard-01-chunk-0">
            <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
              <CardTitle className="text-dark300_light900 text-sm font-medium">
                Total Revenue
              </CardTitle>
              <DollarSign className="text-dark300_light900 size-4" />
            </CardHeader>
            <CardContent>
              <div className="text-dark300_light900 text-2xl font-bold">$45,231.89</div>
              <p className="text-xs text-lime-900">
                +20.1% from last month
              </p>
            </CardContent>
          </Card>
          <Card x-chunk="dashboard-01-chunk-1">
            <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
              <CardTitle className="text-dark300_light900 text-sm font-medium">
                Subscriptions
              </CardTitle>
              <Users className="text-dark300_light900 size-4" />
            </CardHeader>
            <CardContent>
              <div className="text-dark300_light900 text-2xl font-bold">+2350</div>
              <p className="text-xs text-lime-900">
                +180.1% from last month
              </p>
            </CardContent>
          </Card>
          <Card x-chunk="dashboard-01-chunk-2">
            <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
              <CardTitle className="text-dark300_light900 text-sm font-medium">Sales</CardTitle>
              <CreditCard className="text-dark300_light900 size-4" />
            </CardHeader>
            <CardContent>
              <div className="text-dark300_light900 text-2xl font-bold">+12,234</div>
              <p className="text-xs text-lime-900">
                +19% from last month
              </p>
            </CardContent>
          </Card>
          <Card x-chunk="dashboard-01-chunk-3">
            <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
              <CardTitle className="text-sm font-medium">Active Now</CardTitle>
              <Activity className="text-dark300_light900 size-4" />
            </CardHeader>
            <CardContent>
              <div className="text-2xl font-bold">+573</div>
              <p className="text-xs text-lime-900">
                +201 since last hour
              </p>
            </CardContent>
          </Card>
        </div>
        <div className="grid gap-4 md:gap-8 lg:grid-cols-2 xl:grid-cols-2">
          <Card
            className="xl:col-span-2" x-chunk="dashboard-01-chunk-4"
          >
            <CardHeader className="flex flex-row items-center">
              <div className="grid gap-2">
                <CardTitle className="text-dark300_light900">Google Results</CardTitle>
                <CardDescription className="text-dark300_light900">
                  Recent results from your google accounts.
                </CardDescription>
              </div>
            </CardHeader>
            <CardContent>
              <Table className="text-dark300_light900">
                <TableHeader>
                  <TableRow>
                    <TableHead>Account</TableHead>
                    <TableHead>Campaign</TableHead>
                    <TableHead>Ad ID</TableHead>
                    <TableHead>Revenue</TableHead>
                    <TableHead >Ad Cost</TableHead>
                    <TableHead >Profit</TableHead>
                    <TableHead >ROI</TableHead>
                  </TableRow>
                </TableHeader>

                <TableBody>
                  <TableRow>
                    <TableCell>A</TableCell>
                    <TableCell>B</TableCell>
                    <TableCell>C</TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">D</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">E</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">F</Badge>
                    </TableCell>
                    <TableCell>G</TableCell>
                  </TableRow>

                  <TableRow>
                    <TableCell>A</TableCell>
                    <TableCell>B</TableCell>
                    <TableCell>C</TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">D</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">E</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">F</Badge>
                    </TableCell>
                    <TableCell>G</TableCell>
                  </TableRow>

                  <TableRow>
                    <TableCell>A</TableCell>
                    <TableCell>B</TableCell>
                    <TableCell>C</TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">D</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">E</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">F</Badge>
                    </TableCell>
                    <TableCell>G</TableCell>
                  </TableRow>

                </TableBody>
              </Table>
            </CardContent>
          </Card>
        </div>

        <div className="grid gap-4 md:gap-8 lg:grid-cols-2 xl:grid-cols-2">
          <Card
            className="xl:col-span-2" x-chunk="dashboard-01-chunk-4"
          >
            <CardHeader className="flex flex-row items-center">
              <div className="grid gap-2">
                <CardTitle className="text-dark300_light900">Facebook Results</CardTitle>
                <CardDescription className="text-dark300_light900">
                  Recent results from your facebook accounts.
                </CardDescription>
              </div>
            </CardHeader>
            <CardContent>
              <Table className="text-dark300_light900">
                <TableHeader>
                  <TableRow>
                    <TableHead>Account</TableHead>
                    <TableHead>Campaign</TableHead>
                    <TableHead>Ad ID</TableHead>
                    <TableHead>Revenue</TableHead>
                    <TableHead >Ad Cost</TableHead>
                    <TableHead >Profit</TableHead>
                    <TableHead >ROI</TableHead>
                  </TableRow>
                </TableHeader>

                <TableBody>
                  <TableRow>
                    <TableCell>A</TableCell>
                    <TableCell>B</TableCell>
                    <TableCell>C</TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">D</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">E</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">F</Badge>
                    </TableCell>
                    <TableCell>G</TableCell>
                  </TableRow>

                  <TableRow>
                    <TableCell>A</TableCell>
                    <TableCell>B</TableCell>
                    <TableCell>C</TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">D</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">E</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">F</Badge>
                    </TableCell>
                    <TableCell>G</TableCell>
                  </TableRow>

                  <TableRow>
                    <TableCell>A</TableCell>
                    <TableCell>B</TableCell>
                    <TableCell>C</TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">D</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">E</Badge>
                    </TableCell>
                    <TableCell>
                      <Badge className="text-xs" variant="outline">F</Badge>
                    </TableCell>
                    <TableCell>G</TableCell>
                  </TableRow>

                </TableBody>
              </Table>
            </CardContent>
          </Card>
        </div>

        <div>
          <Card>
            <CardHeader className="flex flex-col items-stretch space-y-0 border-b p-0 sm:flex-row">
              <div className="flex flex-1 flex-col justify-center gap-1 px-6 py-5 sm:py-6">
                <CardTitle className="text-dark300_light900">Bar Chart - Interactive</CardTitle>
                <CardDescription className="text-dark300_light900">
                  Showing total Leads for the last 3 months
                </CardDescription>
              </div>
              <div className="flex">
                {["google", "facebook"].map((key) => {
                  const chart = key as keyof typeof chartConfig
                  return (
                    <button
                      key={chart}
                      data-active={activeChart === chart}
                      className="data-[active=false]:text-dark300_light900 relative z-30 flex flex-1 flex-col justify-center gap-1 border-t px-6 py-4 text-left even:border-l data-[active=true]:bg-primary-100 sm:border-l sm:border-t-0 sm:px-8 sm:py-6"
                      onClick={() => setActiveChart(chart)}
                    >
                      <span className="text-xs text-primary-500">
                        {chartConfig[chart].label}
                      </span>
                      <span className="text-lg font-bold leading-none sm:text-3xl">
                        {total[key as keyof typeof total].toLocaleString()}
                      </span>
                    </button>
                  )
                })}
              </div>
            </CardHeader>
            <CardContent className="px-2 sm:p-6">
              <ChartContainer
                config={chartConfig}
                className="aspect-auto h-[250px] w-full"
              >
                <BarChart
                  accessibilityLayer
                  data={chartData}
                  margin={{
                    left: 12,
                    right: 12,
                  }}
                >
                  <CartesianGrid vertical={false} />
                  <XAxis
                    dataKey="date"
                    tickLine={false}
                    axisLine={false}
                    tickMargin={8}
                    minTickGap={32}
                    tickFormatter={(value) => {
                      const date = new Date(value)
                      return date.toLocaleDateString("en-US", {
                        month: "short",
                        day: "numeric",
                      })
                    }}
                  />
                  <ChartTooltip
                    content={
                      <ChartTooltipContent
                        className="w-[150px] bg-white"
                        nameKey="views"
                        labelFormatter={(value) => {
                          return new Date(value).toLocaleDateString("en-US", {
                            month: "short",
                            day: "numeric",
                            year: "numeric",
                          })
                        }}
                      />
                    }
                  />
                  <Bar dataKey={activeChart} fill={`var(--color-${activeChart})`} />
                </BarChart>
              </ChartContainer>
            </CardContent>
          </Card>
        </div>
      </main>
    </div>
  )
}`

Error:
.next/types/app/(root)/(home)/page.ts:28:13
Type error: Type 'OmitWithTag<HTMLAttributes<HTMLDivElement>, keyof PageProps, "default">' does not satisfy the constraint '{ [x: string]: never; }'.
  Property 'color' is incompatible with index signature.
    Type 'string' is not assignable to type 'never'.

`26 |
  27 | // Check the prop type of the entry function
> 28 | checkFields<Diff<PageProps, FirstArg<TEntry['default']>, 'default'>>()
     |             ^
  29 |
  30 | // Check the arguments and return type of the generateMetadata function
  31 | if ('generateMetadata' in entry) {`

**Comment by austin495** [comment]
@icyJoseph Bro can you please help me?

**Reply by icyJoseph** [reply]
Sure.

The easiest is to just:

- rename your page.tsx file to home.tsx 
- then create a page.tsx, again, where you import from home 
- then create a component without props, that renders what you import from home
- maybe you need to make the className prop optional 

This assumes that, no params exists in the path to this page.tsx

In short, the problem is that, pages in app router, are passed props, by the framework, and the props the framework will pass to your component, are not matching what you define as your component's signature. That is the class name prop.

**Reply by austin495** [reply]
Bro anothe issue is `4:17  Error: Identifier 'Space_Grotesk' is not in camel case.  camelcase` on `npx next build`

./app/laout.tsx:
`import { ThemeProvider } from "@/context/ThemeProvider";
import { ClerkProvider } from "@clerk/nextjs";
import type { Metadata } from "next";
import { Inter, Space_Grotesk } from "next/font/google";
import React from "react";
import "./globals.css";

const inter = Inter({
  subsets: ["latin"],
  weight: ['100', '200', '300', '400', '500', '600', '700', '800', '900'],
  variable: '--font-inter'
});

const spaceGrotesk = Space_Grotesk({
  subsets: ["latin"],
  weight: ['300', '400', '500', '600', '700'],
  variable: '--font-spaceGrotesk'
});

export const metadata: Metadata = {
  title: "Marketing Analytics",
  description: "Generated by create next app",
};

export default function RootLayout({ children, }: Readonly<{ children: React.ReactNode; }>) {
  return (

    <html lang="en">
      <body className={`${inter.variable} ${spaceGrotesk.variable}`}>
        <ClerkProvider
          appearance={{
            elements: {
              formButtonPrimary: 'primary-gradient',
              footerActionLink: 'primary-text-gradient hover:text-primary-500'
            }
          }}
        >
          <ThemeProvider>
            {children}
          </ThemeProvider>
        </ClerkProvider>
      </body>
    </html>

  );
}
`

**Reply by icyJoseph** [reply, marked as answer]
Hmmm that's a lint issue... Maybe try:

```ts
import { Inter, Space_Grotesk as SpaceGrotesk} from "next/font/google";
```

And then use `SpaceGrotesk` in your code.

**Reply by icyJoseph** [reply]
Did you resolve your issues?

**Reply by austin495** [reply]
Yes
